inventory/any.cf

```diff
@@ -141,22 +141,43 @@ bundle agent cfe_autorun_inventory_memory
 # @brief Inventory the memory (Enterprise only)
 {
   vars:
-    enterprise.!aix::
-      # due to a Windows issue this is set to 0 there for now
-      "total" string => ifelse("windows", "0",
-                               $(mon.value_mem_total)),
-      meta => { "inventory", "attribute_name=Memory size (MB)" };
-
-    enterprise::
-      "free" string => ifelse("windows", "0",
-                              $(mon.value_mem_free)),
-      meta => { "report" };
-
+@if minimum_version(3.11)
+      # The `with` attribute is necessary for this to work in a single promise.
+    enterprise.windows::
+      "total"
+        meta => { "inventory", "attribute_name=Memory size (MB)" },
+        string => format( "%d", eval("$(with)/1024", "math", "infix" )),
+        if => not( isvariable( "total" ) ),
+        with => nth( string_split( execresult("wmic OS get TotalVisibleMemorySize /format:list", useshell ),
+                                   "=", 2), 1);
+      "totalPhysical"
```

Comment:
A comment on the first one would be nice. Like:

```
# wmic returns "TotalVisibleMemorySize=10760224" so split on = and take the second item (0-based with nth())
```

Reply:
ACK